vtStorAtaProtocol/Platform/Linux/ProtocolAtaPassThrough.h

```diff
@@ -30,33 +30,116 @@ namespace vtStor
+    U8                HighFeature;            // Byte 3
+    U8                LowFeature;             // Byte 4
+    U8                HighSectorCount;        // Byte 5
+    U8                LowSectorCount;         // Byte 6
+    U8                HighObLbaLow;           // Byte 7
+    U8                LowObLbaLow;            // Byte 8
+    U8                HighObLbaMid;           // Byte 9
+    U8                LowObLbaMid;            // Byte 10
+    U8                HighObLbaHigh;          // Byte 11
+    U8                LowObLbaHigh;           // Byte 12
+    U8                LowDevice;              // Byte 13
+    U8                LowCommand;             // Byte 14
+    U8                Control;                // Byte 15
+};
+
 class VT_STOR_ATA_PROTOCOL_API cAtaPassThrough : public cProtocolInterface
 {
 
 public:
-    //virtual eErrorCode IssueCommand( const DeviceHandle& Handle, std::shared_ptr<const cBufferInterface> Essense, std::shared_ptr<cBufferInterface> DataBuffer ) override;
+    virtual eErrorCode IssueCommand(const DeviceHandle& Handle, std::shared_ptr<const cBufferInterface> Essense, std::shared_ptr<cBufferInterface> DataBuffer) override;
 
 private:
-    void InitializePassThroughDirect(
-        const StorageUtility::Ata::sCommandCharacteristic& CommandCharacteristics,
-        const StorageUtility::Ata::uTaskFileRegister& PreviousTaskFile,
-        const StorageUtility::Ata::uTaskFileRegister& CurrentTaskFile,
-        std::shared_ptr<cBufferInterface> DataBuffer,
-        U32 TimeoutValueInSeconds
-        );
-
-    //! Initialize the ATA flags in the ATA_PASS_THROUGH_DIRECT structure
-    //!
-    void InitializeFlags( const StorageUtility::Ata::sCommandCharacteristic& AtaCommandCharacteristic );
+    void InitializeCdbRegisters(const StorageUtility::Ata::uTaskFileRegister& PreviousTaskFile, const StorageUtility::Ata::uTaskFileRegister& CurrentTaskFile);
 
-    //! Initialize the iCurrentTaskFile. and the PreviousTaskFile in the ATA_PASS_THROUGH_DIRECT structure
-    //!
-    void InitializeTaskFileInputRegisters( const StorageUtility::Ata::uTaskFileRegister& PreviousTaskFile, const StorageUtility::Ata::uTaskFileRegister& CurrentTaskFile );
+    void cAtaPassThrough::InitializePassThroughDirect(
+            const StorageUtility::Ata::sCommandCharacteristic& CommandCharacteristics,
+            const StorageUtility::Ata::uTaskFileRegister& PreviousTaskFile,
+            const StorageUtility::Ata::uTaskFileRegister& CurrentTaskFile,
+            std::shared_ptr<cBufferInterface> DataBuffer, U32 TimeoutValueInSeconds
+            );
 
-    //eErrorCode IssuePassThroughDirectCommand( const DeviceHandle& Handle, U32& BytesReturned );
+    void InitializeFlags(const StorageUtility::Ata::sCommandCharacteristic& AtaCommandCharacteristic);
 
+    eErrorCode IssuePassThroughDirectCommand(const int &FileDescriptor);
 private:
-   // ATA_PASS_THROUGH_DIRECT m_AtaPassThrough;
+    struct sg_io_hdr m_ATAPassThrough;

```

Comment:
Cdb16 is "command descriptor block" 16 byte. It is a term of Scsi, i think we can use for simple name.